client/example_test.go

```diff
@@ -0,0 +1,257 @@
+	}
+	defer simulator.Close()
+
+	srk, err := client.StorageRootKeyECC(simulator)
+	if err != nil {
+		log.Fatalf("failed to create storage root key: %v", err)
+	}
+
+	sealedSecret := []byte("password to seal")
+
+	sel := tpm2.PCRSelection{Hash: tpm2.AlgSHA256, PCRs: []int{7}}
+	// Seal the data to the current value of PCR7.
+	sealedBlob, err := srk.Seal([]byte(sealedSecret), client.SealCurrent{PCRSelection: sel})
+	if err != nil {
+		log.Fatalf("failed to seal to SRK: %v", err)
+	}
+
+	// Validate by unsealing the sealed blob. Because it is possible that a TPM can seal a secret
+	// properly but fail to certify it (thus we shouldn't unseal it because the creation status
+	// cannot be verify). This ensures we can unseal the sealed blob, and that its contents are
+	// equal to what we sealed.
+	_, err = srk.Unseal(sealedBlob, client.CertifyCurrent{PCRSelection: sel})
+	if err != nil {
+		// TODO: handle unseal error.
+		log.Fatalf("failed to unseal blob: %v", err)
+	}
+	// TODO: use unseal output.
+}
+
+func ExampleKey_GetSigner() {
+	// TODO: use real TPM.
+	simulator, err := simulator.Get()
+	if err != nil {
+		log.Fatalf("failed to initialize simulator: %v", err)
+	}
+	defer simulator.Close()
+	key, err := client.NewKey(simulator, tpm2.HandleOwner, signerTemplateECC())
+	if err != nil {
+		log.Fatalf("failed to create signing key: %v", err)
+	}
+	defer key.Close()
+
+	toSign := []byte("message to sign")
+	hash := hashAlg.New()
+	hash.Write(toSign)
+	digest := hash.Sum(nil)
+
+	cryptoSigner, err := key.GetSigner()
+	if err != nil {
+		log.Fatalf("failed to create crypto signer: %v", err)
+	}
+	sig, err := cryptoSigner.Sign(nil, digest, hashAlg)
+	if err != nil {
+		log.Fatalf("failed to sign: %v", err)
+	}
+
+	// Verifier needs to establish trust in signer.Public() (via a certificate,
+	// TPM2_ActivateCredential, TPM2_Certify).
+	if !ecdsa.VerifyASN1(cryptoSigner.Public().(*ecdsa.PublicKey), digest, sig) {
+		// TODO: handle signature verification failure.
+		log.Fatal("failed to verify digest")
+	}
+}
+
+func ExampleKey_SignData() {
+	// TODO: use real TPM.
+	simulator, err := simulator.Get()
+	if err != nil {
+		log.Fatalf("failed to initialize simulator: %v", err)
+	}
+	defer simulator.Close()
+	key, err := client.NewKey(simulator, tpm2.HandleOwner, signerTemplateECC())
+	if err != nil {
+		log.Fatalf("failed to create signing key: %v", err)
+	}
+	defer key.Close()
+
+	toSign := []byte("message to sign")
+	hash := hashAlg.New()
+	hash.Write(toSign)
+	digest := hash.Sum(nil)
+
+	sig, err := key.SignData(toSign)
+	if err != nil {
+		log.Fatalf("failed to sign data: %v", err)
+	}
+
+	// Verifier needs to establish trust in signer.Public() (via a certificate,
+	// TPM2_ActivateCredential, TPM2_Certify).
+	if !ecdsa.VerifyASN1(key.PublicKey().(*ecdsa.PublicKey), digest, sig) {
+		// TODO: handle signature verification failure.
+		log.Fatal("failed to verify digest")
+	}
+}
+
+func signerTemplateECC() tpm2.Public {
```

Comment:
I'm not sure we want this there as it's one random template for ECC signing; it's not a template in the spec (like the EK), or even a commonly used one (like AK ECC or AK RSA).

Reply:
That makes sense. Now I have another question, I think this helper method should appear in the example doc as well (since it was referenced in the above example, but I didn't see on my godoc server. And usually the example should be wrapped in a main func, but it didn't do that on my local godoc sever as well. I am not sure if it is my godoc issue. If it renders correctly on your end we can resolve this.
<img width="853" alt="Screen Shot 2021-08-03 at 1 49 23 PM" src="https://user-images.githubusercontent.com/14238798/128084872-ea2e3f89-54df-4fad-b86d-c699b6746f4b.png">

Reply:
Yes, this is a great point. I understand your earlier comment now. I'm not sure how to include that function in the generated godoc example. Other options include: 1. copy-pasting that template into the example code itself 2. a comment (rendering the code not working as is) or 3. using the AK template but removing restricted st. it can sign non-TPM generated data 4. leaving as is.

Reply:
Another option: `func Example_SignerTemplateECC()`. This allows users to see what a signer template might look like in the godocs.

I prefer 1 and this new option.

Reply:
For the second concern about running examples in the go playground, you can do `godoc -http=:6060 -play=True`. However, these examples aren't meant to run on the playground as they interact with the event log. Furthermore, the go playground actually compiles and builds the code using the latest tagged release. That runs into problems like this (due to tpm2tools -> client rename): 
```
prog.go:5:2: module github.com/google/go-tpm-tools@latest found (v0.2.1), but does not contain package github.com/google/go-tpm-tools/client
```

Reply:
I went with option 1 btw. PTAL

Reply:
Thanks! First option LGTM.